### PR TITLE
Add config to disable html area sanitizing #9638

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/processor/HtmlAreaContentProcessor.java [hunk 1]
```diff
@@ -58,10 +59,14 @@
 
 import static com.google.common.base.Strings.nullToEmpty;
 
-@Component
+@Component(configurationPid = "com.enonic.xp.admin")
```
Comment (anatol-sialitski) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/processor/HtmlAreaContentProcessor.java:62]: use `configurationPid = "com.enonic.xp.content"` instead

File: modules/core/core-content/src/test/java/com/enonic/xp/core/impl/content/processor/HtmlAreaContentProcessorTest.java [hunk 2]
```diff
@@ -111,16 +112,16 @@ public void setUp()
         Mockito.when( contentTypeService.getByName( params ) ).thenReturn( contentType );
         Mockito.when( xDataService.getByNames( Mockito.isA( XDataNames.class ) ) ).thenReturn( XDatas.empty() );
 
-        HtmlSanitizer htmlSanitizer = new HtmlSanitizerImpl();
+        contentConfig = Mockito.mock( ContentConfig.class );
```
Comment (anatol-sialitski) [inline modules/core/core-content/src/test/java/com/enonic/xp/core/impl/content/processor/HtmlAreaContentProcessorTest.java:115]: You can write that instead
```
ContentConfig contentConfig = mock( ContentConfig.class, invocation -> invocation.getMethod().getDefaultValue() );
```

**Review by rymsha — CHANGES_REQUESTED** [review]
also double check why sanitizer replaces nbsp with non-breaking

File: modules/runtime/src/home/config/com.enonic.xp.content.cfg [hunk 3]
```diff
@@ -1,3 +1,5 @@
 #auditlog.enabled = true
 
 #attachments.allowUnsafeNames = false
+
+#htmlarea.sanitizing.enabled = true
```
Comment (rymsha) [inline modules/runtime/src/home/config/com.enonic.xp.content.cfg:5]: change dafault to `false` and don't mention it in config file

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/runtime/src/home/config/com.enonic.xp.content.cfg [hunk 4]
```diff
@@ -1,3 +1,5 @@
 #auditlog.enabled = true
 
 #attachments.allowUnsafeNames = false
+
+#htmlarea.sanitizing.enabled = false
```
Comment (rymsha) [inline modules/runtime/src/home/config/com.enonic.xp.content.cfg:5]: delete this line

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/processor/InternalHtmlSanitizer.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        .allowStandardUrlProtocols()
+        .allowStyling()
+        .toFactory();
+
+    private InternalHtmlSanitizer()
+    {
+
+    }
+
+    public static String sanitize( final String value, final boolean strict )
```
Comment (rymsha) [inline modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/processor/InternalHtmlSanitizer.java:64]: Use two factory methods instead

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**